src/components/switch/switch.test.js

```diff
@@ -540,4 +519,264 @@ context("Testing Switch component", () => {
+      CypressMountWithProviders(<SwitchComponent name={CHARACTERS.STANDARD} />);
+
+      cy.checkAccessibility();
+    });
+
+    // FE-4678
+    describe.skip("should check Switch component validation accessibility", () => {
+      it("check Switch component accessibility with validation", () => {
+        CypressMountWithProviders(<SwitchComponentValidations />);
+
+        cy.checkAccessibility();
+      });
+
+      it("check Switch component accessibility with validation on label", () => {
+        CypressMountWithProviders(
+          <SwitchComponentValidations validationOnLabel />
+        );
+
+        cy.checkAccessibility();
+      });
+    });
+
+    it.each([true, false])(
+      "check Switch component accessibility reverse set to %s",
+      (boolVal) => {
+        CypressMountWithProviders(<SwitchComponent reverse={boolVal} />);
+
+        cy.checkAccessibility();
+      }
+    );
+
+    it.each([SIZE.SMALL, SIZE.LARGE])(
+      "check Switch component accessibility with size set to %s",
+      (size) => {
+        CypressMountWithProviders(<SwitchComponent size={size} />);
+
+        cy.checkAccessibility();
+      }
+    );
+
+    it("check Switch component accessibility with value prop", () => {
+      CypressMountWithProviders(<SwitchComponent value="switchvalue" />);
+
+      cy.checkAccessibility();
+    });
+
+    it.each([399, 400, 401])(
+      "check Switch component accessibility when label is inline with adaptiveLabelBreakpoint %s",
+      (breakpoint) => {
+        cy.viewport(400, 300);
+
+        CypressMountWithProviders(
+          <SwitchComponent labelInline adaptiveLabelBreakpoint={breakpoint} />
+        );
+
+        cy.checkAccessibility();
+      }
+    );
+
+    it("check Switch component accessibility as a required field", () => {
+      CypressMountWithProviders(<SwitchComponent required />);
+
+      cy.checkAccessibility();
+    });
+
+    it.each(["bottom", "left", "right", "top"])(
+      "check Switch component accessibility with tooltip positioned to the %s",
+      (position) => {
+        CypressMountWithProviders(
+          <Box m="250px">
+            <SwitchComponent
+              labelHelp="Switch info"
+              tooltipPosition={position}
+            />
+          </Box>
+        );
+
+        cy.checkAccessibility();
+      }
+    );
+
+    it("check Switch component accessibility with helpAriaLabel", () => {
+      CypressMountWithProviders(
+        <SwitchComponent
+          labelHelp="Label Help"
+          helpAriaLabel="This text provides more information for the label"
+        />
+      );
+
+      cy.checkAccessibility();
+    });
+
+    it.each([true, false])(
+      "check Switch component accessibility with defaultChecked %s",
+      (boolVal) => {
+        CypressMountWithProviders(
+          <Switch label="Label" name="switch-name" defaultChecked={boolVal} />
+        );
+
+        cy.checkAccessibility();
+      }
+    );
```

Comment:
You could add a test for the 'WithMargin' story which is in switch.stories.tsx. It might also be worth adding tests for the number of Validation stories included in switch.stories.tsx, though they would probably have to be skipped.